### PR TITLE
Rename `Region` views

### PR DESCRIPTION
**Pull request by lmondada**
I propose the following renaming
 - `FlatRegionView` -> `SiblingGraph`
 - `RegionView` -> `DescendantsGraph`
 - `Region` -> `HierarchyView`

I've reserved the term `View` for the trait rather than the `struct`s as that is closer to our use of the word elsewhere. Happy to discuss other opinions.

In addition to being more descriptive, this will also allow us to make the complementarity with #336 more obvious.

### CODE REVIEW

**Comment by lmondada** [comment]
Credits to @acl-cqc for the suggestions!

File: src/hugr/hierarchy.rs [hunk 1]
```diff
@@ -1,14 +1,16 @@
-//! Region-level views of a HUGR.
```
Comment (ss2165) [inline src/hugr/hierarchy.rs:1]: maybe the file/module should be called `views.rs`?

**Review by ss2165 — COMMENTED** [review]
I like the names, and the reserving of `View` for traits. Just a suggestion on the module name.

File: src/hugr/hierarchy.rs (hunk 1 above)
Comment (lmondada) [reply src/hugr/hierarchy.rs:1]: I would like that, but there is already the `view.rs` file in the same folder that contains `HugrView` etc.

I could make a `views` folder with
- `views/hugr.rs`
- `views/hierarchy.rs`
in it?

Comment (ss2165) [reply src/hugr/hierarchy.rs:1]: or just go with `hierarchical_views`

**Review by ss2165 — APPROVED** [review]
👍 thanks

**Pull request merged**